mailet/base/src/test/java/org/apache/mailet/base/AutomaticallySentMailDetectorImplTest.java

```diff
@@ -286,4 +288,24 @@ public void isMdnSentAutomaticallyShouldManageItsMimeType() throws Exception {
         assertThat(new AutomaticallySentMailDetectorImpl().isMdnSentAutomatically(fakeMail)).isFalse();
     }
 
+    @Test
+    public void isMdnSentAutomaticallyShouldNotThrowOnBodyPartsWithManyLines() throws Exception {
+        int mime4jDefaultMaxHeaderCount = 1000;
+        int headerCount = mime4jDefaultMaxHeaderCount + 10;
+        MimeMessage message = MimeMessageBuilder.mimeMessageBuilder()
+            .addHeaders()
+            .setMultipartWithBodyParts(MimeMessageBuilder.bodyPartBuilder()
+                .addHeaders(Collections.nCopies(headerCount, new MimeMessageBuilder.Header("name", "value")))
+                .data("The body part have 1010 headers, which overpass MIME4J default limits")
+                .build())
+            .build();
+
+        FakeMail fakeMail = FakeMail.builder()
+            .sender(MailAddressFixture.ANY_AT_JAMES)
+            .mimeMessage(message)
+            .build();
+
+        assertThat(new AutomaticallySentMailDetectorImpl().isMdnSentAutomatically(fakeMail)).isFalse();
```

Comment:
It stands for https://tools.ietf.org/html/rfc3798

Message Disposition Notification